### ISSUE TITLE
How to have modals in separate template files?

### ISSUE THREAD

**Issue by repzilon** [dougludlow/ng2-bs3-modal#120]
I would like to have the modals markup in template files from their caller template file. In other words, I would like to have the following file layout : 

MainScreen.component.ts
MainScreen.html
Modal1.component.ts
Modal1.html
Modal2.component.ts
Modal2.html

where MainScreen calls Modal1 and Modal2 when needed. I don't see how I can acheive that from your Readme file.

Also, how can I can call a modal from a modal?

**Comment by dougludlow** [comment]
@repzilon you could create a component that wraps the modal component and then reuse it wherever needed. 

In the project's [demo](http://dougludlow.github.io/ng2-bs3-modal/), you can see an example of opening another modal from a modal (click on the the "nested modals" button). Feel free to check out the [code](https://github.com/dougludlow/ng2-bs3-modal/blob/master/demo/modal-demo.component.html#L66-L79).

**Comment by zsmorgan21** [comment]
@dougludlow Can you please provide more information on @repzilon 's first question? How would the modal components be loaded dynamically without leaving the "MainScreen" component?

**Comment by dougludlow** [comment]
@zsmorgan21 I'm not sure what you're asking exactly, could you elaborate?

**Comment by beneschmidt** [comment]
Simply put the modal logic and templating into seperated files (just like stated in the question). From your **MainScreen.html** page, you can use the component just as any other component with the element tag, something likes this:
`<modal1 #modal1></modal1>`
Notice the _modal1_ attribute name. By giving the component this identifier, you can call public functions of your modal component. For example if you have a button that should open the modal on click you could use something like this:
`<button class="btn btn-primary" (click)='modal1.openIt()'>open modal</button>`

Now let's check out the **Modal1.component.ts**. You need the public function _openIt_ that you are calling from the MainScreen.html, so you can finally open the modal that you specified in the **Modal1.component.html**. It probably looks something like this:
`openIt(){
  this.modal.open();
}`
For this to work, you need to reference your modal in the **Modal1.component.ts**, but I think this is documented somewhere in this repo.

To sum it up again:

- put the modal logic and templating into seperated component files
- use the newly created modal component in your calling html page (with the element tag)
- give this component a reference variable (like #modal1)
- call a public function of the reference (for example on button click with `(click)='modal1.publicFunction()'`)
- create this public function in your modal component and there call the _open_ function of the real modal itself

Hope this helps
